src/components/connect-request.js

```diff
@@ -27,62 +45,69 @@ const connectRequest = (mapPropsToConfig, options = {}) => (WrappedComponent) =>
         }
 
         componentDidMount() {
-            this.requestAsync(this.props, false, true);
+            const configs = mapPropsToConfig(this.props);
+            this.requestAsync(configs, false, true);
         }
 
         componentDidUpdate(prevProps) {
-            const prevConfig = mapPropsToConfig(prevProps);
-            const config = mapPropsToConfig(this.props);
-            const prevQueryKey = prevConfig && getQueryKey(prevConfig.url, prevConfig.body);
-            const queryKey = config && getQueryKey(config.url, config.body);
-
-            if (prevQueryKey !== queryKey) {
-                this.cancelPendingRequests();
-                this.requestAsync(this.props, false, true);
+            const prevConfigs = toArray(mapPropsToConfig(prevProps)).filter(Boolean);
+            const configs = toArray(mapPropsToConfig(this.props)).filter(Boolean);
+
+            const { cancelKeys, requestKeys } = diffConfigs(prevConfigs, configs);
+            const requestConfigs = prevConfigs.filter((c) => {
+                return requestKeys.includes(getQueryKey(c.url, c.body));
+            });
+
+            if (cancelKeys.length) {
+                this.cancelPendingRequests(cancelKeys);
+            }
+            if (requestConfigs.length) {
+                this.requestAsync(requestConfigs);
             }
         }
 
         componentWillUnmount() {
-            this.cancelPendingRequests();
+            const cancelKeys = Object.keys(this._pendingRequests);
+            this.cancelPendingRequests(cancelKeys);
         }
 
         getWrappedInstance() {
             return this._wrappedInstance;
         }
 
-        cancelPendingRequests() {
+        cancelPendingRequests(cancelKeys) {
             const { dispatch } = this.context.store;
+            toArray(cancelKeys).filter(Boolean).forEach((queryKey) => {
+                dispatch(cancelQuery(queryKey));
+            });
+        }
 
-            for (const queryKey in this._pendingRequests) {
-                if (this._pendingRequests.hasOwnProperty(queryKey)) {
-                    dispatch(cancelQuery(queryKey));
-                }
-            }
+        requestAsync(configs, force = false, retry = false) {
+            // propsToConfig mapping has happened already
+            toArray(configs).filter(Boolean).forEach((c) => {
+                this._makeRequest(c, force, retry);
+            });
         }
 
-        requestAsync(props, force = false, retry = false) {
+        _makeRequest(config, force, retry) {
```

Comment:
one more ask - can you change this to not be prefixed with an underscore? it's mostly a subjective thing but I'd like to reserve the `_` prefix for instance variables